packages/browser/src/login/oidc/oidcHandlers/AuthorizationCodeWithPkceOidcHandler.ts

```diff
@@ -86,48 +86,45 @@ export default class AuthorizationCodeWithPkceOidcHandler
+        // 'cos we'll need to lookup our session information again when the
+        // browser is redirected back to us (i.e. the OAuth client
+        // application) from the Authorization Server.
+        // We don't want to use our session ID as the OAuth 'state' value, as
+        // that session ID can be any developer-specified value, and therefore
+        // may not be appropriate (since the OAuth 'state' value should really
+        // be an unguessable crypto-random value).
+        // eslint-disable-next-line no-underscore-dangle
+        storage.setForUser(signingRequest.state._id, {
+          sessionId: oidcLoginOptions.sessionId,
+        }),
+
+        // Store our login-process state using the session ID as the key.
+        // Strictly speaking, this indirection from our OAuth state value to
+        // our session ID is unnecessary, but it provides a slightly cleaner
+        // separation of concerns.
+        storage.setForUser(oidcLoginOptions.sessionId, {
           // eslint-disable-next-line no-underscore-dangle
-          storage.setForUser(req.state._id, {
-            sessionId: oidcLoginOptions.sessionId,
-          }),
+          codeVerifier: signingRequest.state._code_verifier,
+          issuer: oidcLoginOptions.issuer.toString(),
+          // The redirect URL is read after redirect, so it must be stored now.
+          redirectUrl: oidcLoginOptions.redirectUrl,
+          dpop: oidcLoginOptions.dpop ? "true" : "false",
+        }),
+      ]);
 
-          // Store our login-process state using the session ID as the key.
-          // Strictly speaking, this indirection from our OAuth state value to
-          // our session ID is unnecessary, but it provides a slightly cleaner
-          // separation of concerns.
-          storage.setForUser(oidcLoginOptions.sessionId, {
-            // eslint-disable-next-line no-underscore-dangle
-            codeVerifier: req.state._code_verifier,
-            issuer: oidcLoginOptions.issuer.toString(),
-            // The redirect URL is read after redirect, so it must be stored now.
-            redirectUrl: oidcLoginOptions.redirectUrl,
-            dpop: oidcLoginOptions.dpop ? "true" : "false",
-          }),
-        ])
-          .then(() => {
-            redirector.redirect(req.url.toString(), {
-              handleRedirect: oidcLoginOptions.handleRedirect,
-            });
-          })
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          .catch((err: unknown) => {
-            // eslint-disable-next-line no-console
-            console.error(err);
-          })
-      );
-    });
+      redirector.redirect(signingRequest.url.toString(), {
+        handleRedirect: oidcLoginOptions.handleRedirect,
+      });
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: unknown) {
+      // eslint-disable-next-line no-console
+      console.error(err);
```

Comment:
By the way (irrelevant to this PR, so feel free to continue anyway): why are we even logging this to the console? Isn't it better to just either let it bubble up (potentially up to the console), or not log it at all if it's not relevant to the developer?

Reply:
That's a good question. Initially, it was meant to capture errors thrown by `oidc-client`, and help us debug them. Just swallowing the error didn't feel right, but I think also bubbling up the error was leaking from the underlying library. I guess we should still thrown an error wrapping up the underlying issue.